### PR TITLE
upgrade lm_eval to 0.4.2 following ITREX

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

upgrade following ITREX

## Description

ITREX is upgrading lm_eval version to 0.4.2 and INC examples using ITREX's evaluate need to be updated.
https://github.com/intel/intel-extension-for-transformers/pull/1462

## Expected Behavior & Potential Risk

pre-ci pass

## How has this PR been tested?

local test

## Dependency Change?

lm_eval upgrade

### CODE REVIEW

**Comment by chensuyue** [comment]
INC examples use released ITREX version, so need to wait next ITREX release.

**Comment by xin3he** [comment]
The extension test: https://inteltf-jenk.sh.intel.com/view/lpot-validation/job/intel-lpot-validation-top-mr-extension/5572/

**Comment by xin3he** [comment]
extension test: 
2x: https://inteltf-jenk.sh.intel.com/view/lpot-validation/job/intel-lpot-validation-top-mr-extension/5573/
3x: https://inteltf-jenk.sh.intel.com/view/lpot-validation/job/intel-lpot-validation-top-mr-extension/5574/

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/run_clm_sparsegpt.py [hunk 1]
```diff
@@ -271,8 +270,8 @@ def parse_args():
         help="Transformers parameter: use the external repo")
     
     # Evaluation config
-    parser.add_argument("--tasks", default=["lambada_openai"],
-        help="Usually chosen with ['lambada_openai','hellaswag','winogrande','piqa']",
+    parser.add_argument("--tasks", default="lambada_openai",
+        type=str, help="tasks list for accuracy validation",
```
Comment (changwangss) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/run_clm_sparsegpt.py:274]: help="tasks list for accuracy validation“，it is not list now.
command usage --tasks "piqa,lambada_openai,hellaswag" with no space.

File: examples/pytorch/nlp/huggingface_models/language-modeling/quantization/llm/run_clm_no_trainer.py [hunk 2]
```diff
@@ -50,8 +50,7 @@
                     help="Pad input ids to max length.")
 parser.add_argument("--calib_iters", default=512, type=int,
                     help="calibration iters.")
-parser.add_argument("--tasks", nargs='+', default=["lambada_openai",
-                                                   "hellaswag", "winogrande", "piqa", "wikitext"],
+parser.add_argument("--tasks", default="lambada_openai,hellaswag,winogrande,piqa,wikitext",
                     type=str, help="tasks list for accuracy validation, text-generation and code-generation tasks are different.")
```
Comment (changwangss) [inline examples/pytorch/nlp/huggingface_models/language-modeling/quantization/llm/run_clm_no_trainer.py:54]: str

File: examples/pytorch/nlp/huggingface_models/language-modeling/quantization/llm/run_clm_no_trainer.py [hunk 3]
```diff
@@ -351,62 +350,82 @@ def eval_func(model):
...
+
+    if args.code_generation:
+        from intel_extension_for_transformers.transformers.llm.evaluation.bigcode_eval import evaluate
+        from transformers import AutoTokenizer
+        tokenizer = AutoTokenizer.from_pretrained(args.model, trust_remote_code=args.trust_remote_code)
+        start = time.time()
+        results = evaluate(
+            model=user_model,
+            tokenizer=tokenizer,
+            tasks=",".join(args.tasks),
```
Comment (changwangss) [inline examples/pytorch/nlp/huggingface_models/language-modeling/quantization/llm/run_clm_no_trainer.py:401]: args.tasks

**Review by changwangss — COMMENTED** [review]
please modify them.

**Pull request merged**